examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -591,6 +591,12 @@ int read_python_stack(struct pt_regs* ctx) {
 
   state->python_stack_prog_call_cnt++;
 
+  // The following case should be unreachable. The test serves as a mandatory hint to the verifier
```

Comment:
This is because, in the `PYTHON_STACK_PROG_CNT - 1`th time this program is called, `stack_len` will be `PYTHON_STACK_FRAMES_PER_PROG * (PYTHON_STACK_PROG_CNT - 1)`, right?

Reply:
(the second line is outdated, gonna update it)

yes. but because this is a separate program, from the point of view of the verifier one can attach it to any arbitrary event, and thus it can be called with arbitrary starting conditions. `stack_len` is bounded to 0-STACK_MAX_LEN only through our own external logic that is invisible to the verifier.

Reply:
ok apparently i removed this comment ?!?! will restore

Reply:
Mhm, just tried to understand your logic